src/Datadog.Trace.ClrProfiler.Native/sig_helpers.cpp

```diff
@@ -0,0 +1,266 @@
+#include "sig_helpers.h"
+
+namespace trace {
+
+bool ParseNumber(PCCOR_SIGNATURE* p_sig, ULONG* number) {
+  ULONG result = CorSigUncompressData(*p_sig, number);
+  if (result == -1) {
+    return false;
+  }
+
+  *p_sig += result;
+  return true;
+}
+
+bool ParseTypeDefOrRefEncoded(PCCOR_SIGNATURE* p_sig) {
+  mdToken type_token;
+  ULONG result;
+  result = CorSigUncompressToken(*p_sig, &type_token);
+  if (result == -1) {
+    return false;
+  }
+
+  *p_sig += result;
+  return true;
+}
+
+bool ParseCustomMod(PCCOR_SIGNATURE* p_sig) {
+  if (**p_sig == ELEMENT_TYPE_CMOD_OPT || **p_sig == ELEMENT_TYPE_CMOD_REQD) {
+    *p_sig += 1;
+    return ParseTypeDefOrRefEncoded(p_sig);
+  }
+
+  return false;
+}
+
+bool ParseOptionalCustomMods(PCCOR_SIGNATURE* p_sig) {
+  for (;;) {
```

Comment:
Non-blocking, but why not a while loop?

Reply:
No good reason, other than the David Broman source I was referencing had this. I can convert it to a while loop for readability

Reply:
Up to you.

Reply:
Replaced with a `while(true)` loop in d2460481325ed2afb8dc02b79f16510a88910217